### PR TITLE
chore: absorb extracted test suite

### LINKED ISSUE #142: Absorb extracted test suite from go-sdk-contrib
### Requirements

As discussed [here](https://github.com/open-feature/flagd/issues/176), the integration test suite is being extracted into go-sdk-contrib. Absorb it in go-sdk.

### PR DESCRIPTION
**Pull request by skyerus**
## This PR

- Absorb extracted test suite from go-sdk-contrib

### Related Issues

Fixes #142

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,6 +9,7 @@ require (
 	github.com/golang/protobuf v1.5.2
 	github.com/open-feature/flagd v0.3.1
 	github.com/open-feature/go-sdk-contrib/providers/flagd v0.1.2
+	github.com/open-feature/go-sdk-contrib/tests/flagd v0.1.0
```
Comment (skyerus) [inline go.mod:12]: TODO: replace once released

**Review by toddbaert — APPROVED** [review]
```
ok  	github.com/open-feature/go-sdk/integration	1.081s	coverage: [no statements]
```

Looks like it's working!

You might want to add a small blurb to the integration testing part of the readme, so people know where to find these tests.

**Comment by skyerus** [comment]
> ```
> ok  	github.com/open-feature/go-sdk/integration	1.081s	coverage: [no statements]
> ```
> 
> Looks like it's working!
> 
> You might want to add a small blurb to the integration testing part of the readme, so people know where to find these tests.

Good idea, done!

**Comment by toddbaert** [comment]
@skyerus could you resolve these conflict and then merge? I'm hoping this could help with https://github.com/open-feature/flagd/pull/312

**Review by james-milligan — APPROVED** [review]

**Pull request merged**